siriuspy/siriuspy/clientarch/time.py

```diff
@@ -1,84 +1,51 @@
+                hour=None, minute=0, second=0, microsecond=0,
+                timestamp=None,
+                timestamp_string=None, timestamp_format=None):
+        """New."""
+        if timestamp:
             if not isinstance(timestamp, float):
                 raise TypeError("Expected an argument of type 'float'.")
-            self.datetime = _datetime.fromtimestamp(timestamp)
-        elif timestamp_string:
+            return Time.fromtimestamp(timestamp)
+        if timestamp_string:
             if not isinstance(timestamp_string, str):
                 raise TypeError("Expected an argument of type 'str'.")
-            self.timestamp_format = '%Y-%m-%d %H:%M:%S.%f' \
-                if timestamp_format is None else timestamp_format
-            self.datetime = _datetime.strptime(
-                timestamp_string, self.timestamp_format)
-        else:
-            self.datetime = _datetime(
-                year=year, month=month, day=day,
-                hour=hour, minute=minute, second=second,
-                microsecond=microsecond)
-
-    @property
-    def year(self):
-        """Return year."""
-        return self.datetime.year
-
-    @property
-    def month(self):
-        """Return month."""
-        return self.datetime.month
-
-    @property
-    def day(self):
-        """Return day."""
-        return self.datetime.day
-
-    @property
-    def hour(self):
-        """Return hour."""
-        return self.datetime.hour
-
-    @property
-    def minute(self):
-        """Return minute."""
-        return self.datetime.minute
-
-    @property
-    def second(self):
-        """Return second."""
-        return self.datetime.second
-
-    @property
-    def microsecond(self):
-        """Return microsecond."""
-        return self.datetime.microsecond
-
-    def get_timestamp(self):
-        """Get timestamp in seconds since epoch format."""
-        return self.datetime.timestamp()
+            return Time.strptime(
+                timestamp_string, '%Y-%m-%d %H:%M:%S.%f'
+                if timestamp_format is None else timestamp_format)
+        return super().__new__(
+            cls, year, month, day, hour, minute, second, microsecond)
 
     def get_iso8601(self):
         """Get iso8601 format."""
-        return self.datetime.astimezone().isoformat()
+        return self.astimezone().isoformat()
 
     def __add__(self, other):
         """Addition."""
-        return self.datetime + other
+        if isinstance(other, _datetime):
+            add = super().__add__(other)
+            return add
+        if isinstance(other, (float, int)):
+            add = super().__add__(_timedelta(seconds=other))
+        else:
+            add = super().__add__(other)
+        return Time(timestamp=add.timestamp())
 
     def __sub__(self, other):
         """Subtraction."""
-        return self.datetime - other
+        if isinstance(other, _datetime):
+            sub = super().__sub__(other)
```

Comment:
there is no need to create a temporary variable here.
`return super().__sub__(other)`

Reply:
done!